kore/src/Kore/Step/Simplification/Condition.hs

```diff
@@ -131,3 +187,168 @@ simplifyPredicate sideCondition predicate = do
             [ "Expecting a \\top term, but found:"
             , unparse conditional
             ]
+
+simplifyConjunctions
+    :: InternalVariable variable
+    => Conditional variable any
+    -> Conditional variable any
+simplifyConjunctions cond =
+    Lens.over
+        (field @"predicate") -- . MultiAnd.lensPredicateSorted sort)
+        simplifyConjunctionByAssumption'
+        cond
+  where
+    sort = predicateSort . predicate $ cond
+    simplifyConjunctionByAssumption' (MultiAnd.fromPredicate -> predicates) =
+        case simplifyConjunctionByAssumption predicates of
+            Unchanged unchanged ->
```

Comment:
1. If `Unchanged`, we should return the original predicate directly.
2. Above, in `fullySimplified`, we should _also_ check that the predicate is simplified.